### PR TITLE
Dashboard providing a collective 'easystats' summary outputs

### LINKED ISSUE #253: Add a dashboard providing a collective 'easystats' summary outputs

### PR DESCRIPTION
**Pull request by IndrajeetPatil**
Closes #253 

TODO:

- [x] Since not all regression model are supported, make the report more fault-tolerant. E.g. a model might be supported in _parameters_, but not in _report_.

### CODE REVIEW

**Comment by strengejacke** [comment]
Is there a way to resize the vertical split? The coefficient plot could be smaller, and the tabular output on the right could be wider. Would be nice to flexibly resize the panels.

File: R/easystats_model_dashboard.R [hunk 1]
```diff
@@ -22,17 +22,20 @@
...
 #' @export
-easystats_model_dashboard <- function(model,
-                                      output_file = "easydashboard.html",
-                                      output_dir = getwd(),
-                                      rmd_dir = system.file("templates/easydashboard.Rmd", package = "easystats")) {
+model_dashboard <- function(model,
+                            output_file = "easydashboard.html",
+                            output_dir = getwd(),
+                            rmd_dir = system.file("templates/easydashboard.Rmd", package = "easystats"),
+                            exponentiate = FALSE) {
```
Comment (IndrajeetPatil) [inline R/easystats_model_dashboard.R]: If we start adding individual parameters like these, we would end up with hundreds of parameters for this function. 

We instead need to have a single parameter per underlying function that we pass to the respective function. 

E.g.

```r
model_dashboard <- function(model,
                            output_file = "easydashboard.html",
                            output_dir = getwd(),
                            rmd_dir = system.file("templates/easydashboard.Rmd", package = "easystats"),
                            parameters.args = list(exponentiate = FALSE, ci = 0.90),
                            performance.args = list(metrics = c("AIC", "AICc", "BIC"), verbose = FALSE) {}
```

Comment (IndrajeetPatil) [reply R/easystats_model_dashboard.R]: @strengejacke Please see my comment here.

Comment (strengejacke) [reply R/easystats_model_dashboard.R]: yes, probably makes sense.

**Comment by strengejacke** [comment]
I like the dashboard! :-)
What do you think about the changes I made?

Try

```r
library(glmmTMB)
  data("fish")
  m1 <- glmmTMB(
    count ~ child + camper + (1 | persons),
    ziformula = ~ child + camper + (1 | persons),
    data = fish,
    family = truncated_poisson()
  )
easystats::model_dashboard(m1)
```

**Comment by strengejacke** [comment]
To summarize the changes a bit:

- I renamed the function from `easystats_model_dashboard` to `model_dashboard`.
- I added `parameters_args` and `performance_args` as arguments, so we have more control (like standardizing, robust SE etc.)
- I changed the table format in the output from plain text to markdown tables.
- I resized the panels for the `check_model()` plot, so it's not too "squeezed", therefore, I updated the `print()` method for `model_performance()`, so that model fit indices are in vertical layout.

**Comment by strengejacke** [comment]
I added a _modelbased_ tab, which tries to use some "clever" heuristic to detect interaction terms. Check out (**after updating** _insight_, `ci_NULL` branch):

```r
data(efc, package = "sjmisc")
efc$c161sex <- sjmisc::to_label(efc$c161sex)
model1 <- lm(neg_c_7 ~ barthtot + c160age + c161sex + I(c160age^2), data = efc)
model2 <- lm(neg_c_7 ~ barthtot + c161sex * c160age + I(c160age^2) * c161sex, data = efc)
model3 <- lm(neg_c_7 ~ c161sex * c160age + I(c160age^2) * c161sex, data = efc)

model_dashboard(model1)
model_dashboard(model2)
model_dashboard(model3)
```

File: inst/templates/easydashboard.Rmd [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+Column {data-width=400}
+-----------------------------------------------------------------------
+
+### Tabular summary
+
+```{r}
+lapply(text_modelbased, function(i) {
+  print_md(i)
+})
+```
```
Comment (strengejacke) [inline inst/templates/easydashboard.Rmd:135]: Now someone needs to find out how to create markdown tables from objects stored in a list.

**Comment by strengejacke** [comment]
For Bayesian models, we have all these sampling-output on top of each page.

**Comment by bwiernik** [comment]
> vertical layout for model_performance()

This is a great idea. We might want to do the same for compare_performance()

**Comment by IndrajeetPatil** [comment]
Thanks, Daniel, for the changes! I think it looks great.

> I added a modelbased tab, which tries to use some "clever" heuristic to detect interaction terms.

Can we add a more informative title to this tab instead of the generic `"Plot"`?  Also, can we leave out the `[[1]]` here?

Screenshot 2022-07-05 at 18 51 25

**Comment by strengejacke** [comment]
> Also, can we leave out the `[[1]]` here?

See https://github.com/easystats/easystats/pull/254#discussion_r913043930

File: inst/templates/easydashboard.Rmd (hunk 2 above)
Comment (bwiernik) [reply inst/templates/easydashboard.Rmd:138]: Either:

(cf. `purrr::walk`)

```r
hidden_lapply <- function(X, FUN, ...) {
  lapply(X, FUN, ...)
  invisible(X)
}
hidden_lapply(text_modelbased, print_md)
```

Or:
```
for (i in text_modelbased) print_md(i)
```

**Comment by bwiernik** [comment]
https://github.com/easystats/easystats/pull/254#discussion_r914917097

**Comment by etiennebacher** [comment]
> > Also, can we leave out the `[[1]]` here?
> 
> See [#254 (comment)](https://github.com/easystats/easystats/pull/254#discussion_r913043930)

Fixed : 

image

**Comment by IndrajeetPatil** [comment]
Awesome! Thanks, @etiennebacher!

File: inst/templates/easydashboard.Rmd [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+Parameter estimates
+=====================================  
+
+Column {data-width=550}
+-----------------------------------------------------------------------
+
+### Plot
+
+```{r dot-whisker}
+plot(table_parameters)
```
[the remark this answers is not in the document]
Comment (etiennebacher) [reply inst/templates/easydashboard.Rmd:92]: for the record, plotly can't convert patchwork to interactive plot, which prevents using it with check_model

**Comment by bwiernik** [comment]
For the tabular summary of predicted values, can we run `format_table()` on it so that the CIs look nice?

**Comment by strengejacke** [comment]
It works for the textual output, but we have no `format()` method when printing in markdown format.

**Comment by strengejacke** [comment]
I think [here](https://github.com/easystats/modelbased/blob/main/R/print.R) we could add a `print_md()` method that calls `format()`, maybe doing some tweaks for markdown output.

**Comment by strengejacke** [comment]
@IndrajeetPatil can you work on your to-do you initially mentioned? Than we might merge this pr, and probably prepare a CRAN submission?

**Comment by IndrajeetPatil** [comment]
@strengejacke Done!

Please check the new behaviour with these two examples, and see if you want to change something.

```r
library(easystats)
library(geepack)
library(performance)

# example-1
fm1 <- aov(breaks ~ wool + tension, data = warpbreaks)
mod <- TukeyHSD(fm1, "tension", ordered = TRUE)
model_dashboard(mod)

# example-2
data(dietox)
dietox$Cu <- as.factor(dietox$Cu)
mf <- formula(Weight ~ Cu * (Time + I(Time^2) + I(Time^3)))
mod <- geeglm(mf, data = dietox, id = Pig, family = poisson("identity"), corstr = "ar1")
model_dashboard(mod)
```

**Comment by strengejacke** [comment]
I improved support for `geeglm` in _performance_. But anyway, should we also use `tryCatch()` for some edge cases, and then also yield a warning? In particular, since _report_ is less stable than the other packages, so a model might be supported for the first three tabs, but does not work for the report-tab.

**Comment by IndrajeetPatil** [comment]
Yes, I have already implemented that. Not by using `tryCatch()`, but by setting `error = TRUE` for each chunk, so that the user also gets to see the exact error that occurred.

**Comment by IndrajeetPatil** [comment]
Please play around with it and try to break it and see how it behaves.

**Comment by strengejacke** [comment]
Ah, ok, that was intentional. Yes, let's go this route!

**Review by strengejacke — APPROVED** [review]

**Comment by strengejacke** [comment]
ok, I made some fixes to _see_ and _modelbased_, so the dashboard should look good now!

**Comment by IndrajeetPatil** [comment]
> ok, I made some fixes to see and modelbased, so the dashboard should look good now!

Something is not right after this change.

If you render the dashboard, you will see that the plot from `check_model()` is printed _twice_ now.

**Comment by strengejacke** [comment]
that should actually be fixed. Have you updated _see_?

**Comment by strengejacke** [comment]
(this should be the fix: https://github.com/easystats/see/commit/d9648c8b9531f801a8a6d32a164cdb26d45951c5)

**Comment by IndrajeetPatil** [comment]
I had updated everything, but maybe I forgot to refresh the R session. 

I will try again.

**Pull request merged**